### ISSUE TITLE
Wrong parsing of encoded newlines in strings

### ISSUE THREAD

**Issue by skuro** [gerritjvv/pjson#31]
## Minimal test case

Given this test JSON file:

```json
["\n"] 
```

When I parse it with `parse-string`:

```clojure
(pjson.core/parse-string (slurp test-file))
```

Expected result:

```clojure
["\n"]  ; <- a newline character
```

Actual result:

    ["\\"]

**Comment by gerritjvv** [comment]
hi, thanks for the report. I'll see if I can find a fix for it over the weekend.

**Comment by skuro** [comment]
Thanks! I updated the description after trying a few other JSON parsers, so that the outcome should be the least surprising one.

**Comment by gerritjvv** [comment]
update:
the issue is in the slowparser logic and was due a a one off bug :(, this has also led me to fix another bug I found during the testing for string escape parsing. I'm down to investigating while one test is failing, this would need to wait till next week though.

FAIL in (test-read-escapes) (escaping_test.clj:38)

If you're use case is urgent please ping me and I'll try to get this through sooner than later, but tbh this week Monday-Friday I'm on holiday :) even from open source.

The changes are:
https://github.com/gerritjvv/pjson/compare/repairunicode?expand=1

**Comment by skuro** [comment]
Thanks for your efforts! I added a PR with what I think fixes the current broken test (plus adding the encoded newline as a test case).

**Comment by gerritjvv** [comment]
great!, thanks, I've merged in your changes and will make a new release. There is still a subtle issue with 
```
JSON.parse("[{\"a\":\"J\\\\\\\"\"}]");
[ { a: 'J\\"' } ]
```
but in pjson its:

```
(pjson/read-str "[{\"a\":\"J\\\\\\\"\"}]")
[{"a" "J\\"}]
```

I do not think this is a huge issue, reason while I'm releasing the current fix, but will take a closer look when I'm back form holidays.

**Comment by gerritjvv** [comment]
done: latest version that contains this fix is: [pjson "0.4.0"].
